manifest-tool times out occasionally

https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=823706

```
2017-06-23T21:48:35.6701850Z -- EXECUTING: manifest-tool --username msimons --password ******** push from-spec manifest.yml
2017-06-23T21:48:42.6187630Z time="2017-06-23T21:48:42Z" level=fatal msg="Inspect of image \"microsoft/dotnet-nightly:1.1.2-sdk-1.0.4-nanoserver\" failed with error: Get https://registry-1.docker.io/v2/: dial tcp: lookup registry-1.docker.io on 8.8.4.4:53: dial udp 8.8.4.4:53: i/o timeout" 
2017-06-23T21:48:42.7191960Z System.InvalidOperationException: Failed to execute manifest-tool --username msimons --password ******** push from-spec manifest.yml
2017-06-23T21:48:42.7210060Z    at Microsoft.DotNet.ImageBuilder.ExecuteHelper.Execute(ProcessStartInfo info, Func`2 executor, Boolean isDryRun, String errorMessage, String executeMessageOverride) in /Microsoft.DotNet.ImageBuilder/ExecuteHelper.cs:line 69
2017-06-23T21:48:42.7226720Z    at Microsoft.DotNet.ImageBuilder.ImageBuilder.PublishManifest() in /Microsoft.DotNet.ImageBuilder/ImageBuilder.cs:line 124
2017-06-23T21:48:42.7241140Z    at Microsoft.DotNet.ImageBuilder.ImageBuilder.Main(String[] args) in /Microsoft.DotNet.ImageBuilder/ImageBuilder.cs:line 45
2017-06-23T21:48:43.2048220Z ##[error]/usr/bin/docker failed with return code: 1
2017-06-23T21:48:43.2098080Z ##[error]/usr/bin/docker failed with error: /usr/bin/docker failed with return code: 1
```

Perhaps retry logic should be added to the image-builder.  Should also consider if the manifest-tool should be more robust.

--- a/src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs
+++ b/src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs
@@ -139,7 +139,10 @@ namespace Microsoft.DotNet.ImageBuilder
 
                         Console.WriteLine($"-- PUBLISHING MANIFEST:{Environment.NewLine}{manifestYml}");
                         File.WriteAllText("manifest.yml", manifestYml.ToString());
-                        ExecuteHelper.Execute(
+
+                        // ExecuteWithRetry because the manifest-tool fails periodically with communicating 
+                        // with the Docker Registry.
+                        ExecuteHelper.ExecuteWithRetry(
                             "manifest-tool",
                             $"--username {Options.Username} --password {Options.Password} push from-spec manifest.yml",
                             Options.IsDryRun);

